kubernetes-client/src/Kubernetes/Client/Config.hs

```diff
@@ -1,136 +1,158 @@
-import           Lens.Micro                 (Lens', lens, set)
-import           Network.Connection         (TLSSettings (..))
-import qualified Network.HTTP.Client        as NH
-import           Network.HTTP.Client.TLS    (mkManagerSettings)
-import           Network.TLS                (Credential, defaultParamsClient)
-import qualified Network.TLS                as TLS
-import qualified Network.TLS.Extra          as TLS
-import           System.Environment         (getEnv)
-import           System.X509                (getSystemCertificateStore)
+{-# LANGUAGE OverloadedStrings #-}
+
+module Kubernetes.Client.Config
+  ( KubeConfigSource(..)
+  , addCACertData
+  , addCACertFile
+  , applyAuthSettings
+  , clientHooksL
+  , Kubernetes.Client.Config.cluster
+  , defaultTLSClientParams
+  , disableServerCertValidation
+  , disableServerNameValidation
+  , disableValidateAuthMethods
+  , kubeClient
+  , loadPEMCerts
+  , newManager
+  , onCertificateRequestL
+  , onServerCertificateL
+  , parsePEMCerts
+  , serviceAccountDir
+  , setCAStore
+  , setClientCert
+  , setMasterURI
+  , setTokenAuth
+  , tlsValidation
+  )
+where
+
+import qualified Kubernetes.OpenAPI.Core as K
+
+import           Control.Applicative                 ((<|>))
+import           Control.Exception.Safe              (MonadThrow, throwM)
+import           Control.Monad.IO.Class              (MonadIO, liftIO)
+import qualified Data.ByteString                     as B
+import qualified Data.ByteString.Base64              as B64
+import qualified Data.ByteString.Lazy                as LazyB
+import           Data.Either.Combinators
+import           Data.Function                       ((&))
+import           Data.Maybe
+import qualified Data.Text                           as T
+import qualified Data.Text.Encoding                  as T
+import qualified Data.Text.IO                        as T
+import           Data.Yaml
+import           Kubernetes.Client.Auth.ClientCert
+import           Kubernetes.Client.Auth.GCP
+import           Kubernetes.Client.Auth.OIDC
+import           Kubernetes.Client.Auth.Token
+import           Kubernetes.Client.Internal.TLSUtils
+import           Kubernetes.Client.KubeConfig
+import           Network.Connection                  (TLSSettings (..))
+import qualified Network.HTTP.Client                 as NH
+import           Network.HTTP.Client.TLS             (mkManagerSettings)
+import qualified Network.TLS                         as TLS
+import           System.Environment                  (getEnv)
+import           System.FilePath
+
+data KubeConfigSource = KubeConfigFile FilePath
+                      | KubeConfigCluster
+
+{-|
+  Creates 'NH.Manager' and 'K.KubernetesClientConfig' for a given
+  'KubeConfigSource'. It is recommended that multiple 'kubeClient' invocations
+  across an application share an 'OIDCCache', this makes sure updation of OAuth
+  token is synchronized across all the different clients being used.
+-}
+kubeClient
```

Comment:
`mkKubeClient` or `mkKubeClientConfig` might be a better name for this, since it's actually creating the client.

Reply:
Called it `mkKubeClientConfig`